### DISCUSSION TITLE
Local build doesn't pickup the lib directory

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#704] [Q&A] [answered]
I have the original Digital "binary installation" at /Applications/Digital/ in MacOS.  Works great and I have the Components|Library items available in the menus when simply starting the app with  a plain ```java -jar /Applications/Digital/Digital.jar```

I've also built the app from (my modified) source and put the created jar file in the same /Applications/Digital/ directory but named as Digital-local.jar

But now when running that in the same manner as the original the 'Library' is not available in the Components menu, nor can loaded projects already having library components in them run.

Just to try something without resorting to reading the source and figuring out what's happening I tried symlinking /Applications/Digital-local to /Applications/Digital/ but no joy.  

I guess I have to sift thru some source this evening... Unless @hneemann beats me to it as usual ^__^

**Comment by hneemann** [comment, marked as answer]
The name of the jar needs to be "Digital.jar". See [here](https://github.com/hneemann/Digital/blob/8255db328c4443540e6de8a0f7f080a588a0aaee/src/main/java/de/neemann/digital/draw/library/ElementLibrary.java#L74). :smile:

**Reply by mengstr** [reply]
Ahhh. The filename is hardcoded into itself. That's kinda ugly, but I'm sure you had good reasons for it.  

So I'll stick with having two separate directories in my /Applications/, one for my version and one for the official. At a total size of a minuscule 20 Megs for the entire installation it really isn't a problem ;-)

Especially if one consider something so simple as Bitwarden (a password manager) is 200 Megs, or Outlook that clocks in at over 2000 Megs

**Reply by hneemann** [reply]
> Ahhh. The filename is hardcoded into itself. That's kinda ugly, but I'm sure you had good reasons for it.

Yes! :smile: The library needs to be found if the simulator is started from within the IDE and if started as a test case. If this would not be the case, developing would be difficult. But I agree: it's not pretty!
